apps/visualization/utils.py

```diff
@@ -267,22 +274,26 @@ def get_unique_countries_data(qs):
             max_indicator_month=Max('indicator_month'),
             indicator_value=F('indicator_value'),
             format=F('format'),
-        ).order_by('subvariable', '-max_indicator_month')
+        ).order_by('-max_indicator_month', 'subvariable')
 
     else:
         all_filters = {
             'emergency': emergency,
         }
         filters = {k: v for k, v in all_filters.items() if v is not None}
+        if region:
+            iso3_list = existing_iso3.filter(region=region).values_list('iso3', flat=True)
+            filters['iso3__in'] = iso3_list
+        else:
+            filters['iso3__in'] = existing_iso3
```

Comment:
We need iso3 value in both case right? Also, can we just filter out the existing_iso3? if yes let's change the existing_iso3 to countries_qs and then let's filter it as following.

```suggestion
        if region:
            countries_qs = existing_iso3.filter(region=region)
        filters['iso3__in'] = countries_qs.values('iso3')
```